### PR TITLE
fix: use execWf when setting PodMetadata

### PR DESCRIPTION
**Pull request by Marks-Ed**
Checklist:

* [ ] My organization is added to [USERS.md](https://github.com/argoproj/argo-workflows/blob/master/USERS.md).

Fixes https://github.com/argoproj/argo-workflows/issues/5942

Tips:

* Your PR needs to pass the required checks before it can be approved. If the check is not required (e.g. E2E tests) it does not need to pass
* Sign-off your commits to pass the DCO check: `git commit --signoff`.
* Run `make pre-commit -B` to fix codegen or lint problems. 
* Say how how you tested your changes. If you changed the UI, attach screenshots.

### CODE REVIEW

File: workflow/controller/workflowpod.go [hunk 1]
```diff
@@ -696,7 +696,7 @@ func isResourcesSpecified(ctr *apiv1.Container) bool {
 
 // addMetadata applies metadata specified in the template
 func (woc *wfOperationCtx) addMetadata(pod *apiv1.Pod, tmpl *wfv1.Template) {
-	if woc.wf.Spec.PodMetadata != nil {
+	if woc.execWf.Spec.PodMetadata != nil {
 		// add workflow-level pod annotations and labels
 		for k, v := range woc.wf.Spec.PodMetadata.Annotations {
```
Comment (sarabala1979) [inline workflow/controller/workflowpod.go:701]: here the another place `woc.wf.Spec.PodMetadata.Annotations` to `woc.execwf.Spec.PodMetadata.Annotations`

File: workflow/controller/workflowpod.go [hunk 2]
```diff
@@ -696,12 +696,12 @@ func isResourcesSpecified(ctr *apiv1.Container) bool {
 
 // addMetadata applies metadata specified in the template
 func (woc *wfOperationCtx) addMetadata(pod *apiv1.Pod, tmpl *wfv1.Template) {
```
Comment (sarabala1979) [inline workflow/controller/workflowpod.go:698]: Can you add a test?

Comment (Marks-Ed) [reply workflow/controller/workflowpod.go:698]: Added tests for setting default and that those defaults are overriden by setting in workflow

**Review by sarabala1979 — APPROVED** [review]

**Comment by Marks-Ed** [comment]
pushed fix for failing lint

**Pull request merged**